tests/vxlan/vnet_utils.py

```diff
@@ -0,0 +1,122 @@
+            combined_template_args.update(arg)
+    except StopIteration:
+        combined_template_args = {}
+
+    rendered = Template(open(path.join("templates",template_name)).read()) \
+                        .render(combined_template_args, **template_kwargs)
+
+    host.copy(content=rendered, dest=dest_file)
+
+def generate_dut_config_files(duthost, mg_facts):
+    """
+    @summary: Generate VNET and VXLAN config files and copy them to DUT.
+    @param duthost: DUT host object
+    @param mg_facts: Minigraph facts
+    """
+    logger.info("Generating config files and copying to DUT")
+
+    vnet_switch_config = [{
+        "SWITCH_TABLE:switch": {
+            "vxlan_port": vc.VXLAN_PORT,
+            "vxlan_router_mac": vc.VXLAN_MAC
+        },
+        "OP": "SET"
+    }]
+
+    duthost.copy(content=json.dumps(vnet_switch_config, indent=4), dest=vc.DUT_VNET_SWITCH_CONFIG)
+
+
+    render_template_to_host("vnet_vxlan.j2", duthost, vc.DUT_VNET_CONF, vc.VNET_CONFIG, mg_facts, ipv6_vxlan_test=vc.IPV6_VXLAN_TEST)
+    render_template_to_host("vnet_interface.j2", duthost, vc.DUT_VNET_INTF_CONFIG, vc.VNET_CONFIG)
+    render_template_to_host("vnet_nbr.j2", duthost, vc.DUT_VNET_NBR_JSON, vc.VNET_CONFIG)
+    render_template_to_host("vnet_routes.j2", duthost, vc.DUT_VNET_ROUTE_CONFIG, vc.VNET_CONFIG, op="SET")
+
+def apply_dut_config_files(duthost):
+    """
+    @summary: Applies config files on disk
+    @param duthost: DUT host object
+    """
+    logger.info("Applying config files on DUT")
+
+    config_files = ["/tmp/vnet.intf.json", "/tmp/vnet.nbr.json"]
+    if vc.APPLY_NEW_CONFIG:
+        config_files.append("/tmp/vnet.conf.json")
+
+    for config in config_files:
+        duthost.shell("sonic-cfggen -j {} --write-to-db".format(config))
+        sleep(3)
+
+    duthost.shell("docker cp {} swss:/vnet.route.json".format(vc.DUT_VNET_ROUTE_CONFIG))
+    duthost.shell("docker cp {} swss:/vnet.switch.json".format(vc.DUT_VNET_SWITCH_CONFIG))
+    if vc.APPLY_NEW_CONFIG:
+        duthost.shell("docker exec swss sh -c \"swssconfig /vnet.switch.json\"")
+        duthost.shell("docker exec swss sh -c \"swssconfig /vnet.route.json\"")
+        sleep(3)
+
+def cleanup_dut_vnets(duthost, mg_facts):
+    """
+    @summary: Removes all VNET information from DUT
+    @param duthost: DUT host object
+    @param mg_factS: Minigraph facts
+    """
+    logger.info("Removing VNET information from DUT")
+
+    for intf in vc.VNET_CONFIG['vlan_intf_list']:
+        duthost.shell("docker exec -i database redis-cli -n 4 del \"VLAN_INTERFACE|{}|{}\"".format(intf['ifname'], intf['ip']))
+
+    for intf in vc.VNET_CONFIG['vlan_intf_list']:
+        duthost.shell("docker exec -i database redis-cli -n 4 del \"VLAN_INTERFACE|{}\"".format(intf['ifname']))
+    
+    for vnet in vc.VNET_CONFIG['vnet_id_list']:
+        duthost.shell("docker exec -i database redis-cli -n 4 del \"VNET|{}\"".format(vnet))
+
+def cleanup_vxlan_tunnels(duthost):
```

Comment:
We would also want to do cleanup of Vlans. This can be ported once ansible fix is available